plugin/http/httpstats/client.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package httpstats
+
+import (
+	"net/http"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"log"
+	"fmt"
+	"context"
+	"time"
+	"strconv"
+	"net/http/httptrace"
+)
+
+var (
+	Started           *stats.MeasureInt64
+	ConnectionsOpened *stats.MeasureInt64
+	Latency           *stats.MeasureFloat64
+
+	StartedCount           *stats.View
+	ConnectionsOpenedCount *stats.View
+	LatencyDistribution    *stats.View
+
+	Host       tag.Key
+	StatusCode tag.Key
+	Path       tag.Key
+	Method     tag.Key
+)
+
+func init() {
+	for _, c := range []struct {
+		measure **stats.MeasureInt64
+		name    string
+		desc    string
+		unit    string
+
+		view **stats.View
+	}{
+		{measure: &Started, name: "started", desc: "Number of HTTP requests started", unit: "1", view: &StartedCount},
+		{measure: &ConnectionsOpened, name: "connections_opened", desc: "Number of HTTP connections opened", unit: "1", view: &ConnectionsOpenedCount},
+	} {
+		fullname := qualify(c.name)
+		m, err := stats.NewMeasureInt64(fullname, c.desc, c.unit)
+		if err != nil {
+			log.Fatalf("failed to create measure %q: %s", fullname, err)
+		}
+		*c.measure = m
+		v, err := stats.NewView(fullname, c.desc, nil, m, &stats.CountAggregation{}, &stats.Cumulative{})
+		if err != nil {
+			log.Fatalf("failed to create view %q: %s", fullname, err)
+		}
+		*c.view = v
+	}
+
+	var err error
+	fullname := qualify("latency")
+	Latency, err = stats.NewMeasureFloat64(fullname, "End-to-end request latency", "microseconds")
+	if err != nil {
+		log.Fatalf("failed to create measure %q: %#v", fullname, err)
+	}
+	aggregation := &stats.DistributionAggregation{1, 100, 1000, 5000, 10000, 50000, 100000, 200000, 500000, 1000000}
```

Comment:
We should investigate what these boundaries should be. We can adopt the gRPC latency boundaries for now.